homeassistant/components/homekit/type_cameras.py

```diff
@@ -155,11 +169,14 @@ async def _async_get_stream_source(self):
         if stream_source:
             return stream_source
         try:
-            return await camera.stream_source()
+            stream_source = await camera.stream_source()
```

Comment:
We should not call methods directly on the camera entity. We should add a helper method to the `camera` integration like `camera.async_get_image` but that returns the stream source

Reply:
Can be done in another PR.

Reply:
Done here https://github.com/home-assistant/core/pull/35704